Fix coverity issues with libarchive/

yaneurabeya opened the pull request.

jsonn commented on libarchive/archive_read_open_filename.c:
```diff
@@ -372,6 +370,10 @@ file_open(struct archive *a, void *client_data)
 
 	return (ARCHIVE_OK);
+done_error:
+	if (0 < fd)
```
No Yoda-style please. Check explicitly for -1.

mmatuska replied:
I do not disagree but I cannot blame neither. @yaneurabeya just kept the style @kientzle used in the file. open() returns -1 on failure. If we want to change this than it should be done thoroughly in all similar cases in the whole file (and there are such cases).

jsonn replied:
At least don't add new instances please.

kientzle replied:
It might make the code clearer to be explicit about both options here:
```
   if (fd == -1 || fd == 0) { /* Don't close a nonexistent fd, don't close stdin. */
   }
```

yaneurabeya replied:
Sounds good -- I'll do that.

jsonn commented on libarchive/archive_write_set_format_iso9660.c:
```diff
@@ -2553,7 +2553,7 @@ set_date_time(unsigned char *p, time_t t)
 set_date_time_null(unsigned char *p)
 {
-	memset(p, '0', 16);
+	memset(p, (int)'0', 16);
```
This is silly and plainly wrong. The semantics of memset are specified to operate on characters and the use of int is only present for the historical reasons of implicit argument promotion.

mmatuska replied:
But I guess Coverity doesn't see it that way :-)

jsonn replied:
Sounds like something to fill a bug report about. There really is absolutely no reason for the cast.

yaneurabeya replied:
The warning was as follows: <pre> 1017786 Memset fill value of '0'

Nominate Defect...
gift  gift  gift
The buffer will be filled with ASCII character '0' instead of actual zero bytes.
In set_date_time_null: A memset fill value of ASCII character '0' is likely intended to be 0 (CWE-665)</pre>

It's warning against valid concerns (noob programmers who don't know how C works might use `'0'` out of ignorance when they meant `0`..

I could ignore the issue in the FreeBSD Coverity runs, but then everyone who runs Coverity against libarchive would need to do the same (NetBSD, Linux variants, etc).

kientzle replied:
I agree that Coverity is probably being overly aggressive here, but an explicit marker to indicate that "yes, we really did mean '0' here" is low cost.

yaneurabeya commented:
@jsonn: I think I've addressed your concerns. Please let me know if there's anything else that needs to be done for this to be merged into the libarchive project. Thanks! -Ngie

kientzle commented:
Thank you!
